### PR TITLE
Another batch of small changes

### PR DESCRIPTION
**Pull request by teo-milea**
## Description
Changes:
* added `dataset.replace_asset`
* simplify project creation
* created uploader field for `Dataset` and `Asset` and defaulted to the logged in user
* added warning when using mock data in `request.code.unsafe_function`
* added domain and execution node specification on requests
* updated 'exchange_credentials_with` permission

## How has this been tested?
- notebooks

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: packages/hagrid/hagrid/orchestra.py [hunk 1]
```diff
@@ -364,6 +364,11 @@ def deploy_to_container(
     env = os.environ.copy()
     env["USER"] = USER
 
+    # stdlib
+    import sys
+
+    print(" ".join(commands), file=sys.stderr)
```
Comment (koenvanderveen) [inline packages/hagrid/hagrid/orchestra.py]: is this debugging code?

Comment (teo-milea) [reply packages/hagrid/hagrid/orchestra.py]: omg yes sorry about that, will remove it

**Review by koenvanderveen — APPROVED** [review]

File: packages/syft/src/syft/service/dataset/dataset.py [hunk 2]
```diff
@@ -597,6 +601,9 @@ def add_asset(
             message=f"Asset '{asset.name}' added to '{self.name}' Dataset."
         )
 
+    def replace_asset(self, asset: CreateAsset):
```
Comment (shubham3121) [inline packages/syft/src/syft/service/dataset/dataset.py:604]: I think we wanted it to be at the Dataset class instead, where it calls the dataset update service. 
The use case is to replace an already uploaded asset.

File: packages/syft/src/syft/service/user/user_service.py [hunk 3]
```diff
@@ -175,6 +175,20 @@ def search(
...
 
+    @service_method(
+        path="user.get_current_user", name="get_current_user", roles=GUEST_ROLE_LEVEL
+    )
+    def get_current_user(self, context: AuthedServiceContext) -> UserView:
+        result = self.stash.get_by_verify_key(
+            credentials=context.credentials, verify_key=context.credentials
+        )
+        if result.is_ok():
+            # this seems weird that we get back None as Ok(None)
```
Comment (shubham3121) [inline packages/syft/src/syft/service/user/user_service.py:186]: It returns None if no user if found with given verify key

Comment (teo-milea) [reply packages/syft/src/syft/service/user/user_service.py:186]: so should we return an error?

Comment (shubham3121) [reply packages/syft/src/syft/service/user/user_service.py:186]: Yes, we can return an error at the user service level,

**Pull request merged**